### ISSUE TITLE
distance calculations do not complete and no .wait() method

### ISSUE THREAD

**Issue by theo-allnutt-bioinformatics** [cogent3/cogent3#938]
Hi,

trying to calculate distances on many separate sequence alignments in a loop. If the there is not way to wait for the calculation to complete, how can it be used on more than one alignment?

e.g.

	cds=fasta2dict(f2) #makes dictionary of all sequences
	for i in f1:
		paralogs={}
		k=i.split("\t")
		k[-1]=k[-1].rstrip("\n")
		name1=k[0]
		paraloglist=k[2:] #list of paralogs to retrieve and align
		for j in paraloglist:
			paralogs[j]=cds[j]
		seqs = make_unaligned_seqs(data=paralogs, moltype="protein")
		aligned = aa_aligner(seqs)
		d = aligned.distance_matrix(calc="hamming", show_progress=False)
		print(d)

Traceback (most recent call last):
  File "/g/data/nm31/scripts/paralog-dists.py", line 106, in <module>
    getgendists(f1,f2,g1)
  File "/g/data/nm31/scripts/paralog-dists.py", line 80, in getgendists
    d = aligned.distance_matrix(calc="hamming", show_progress=False)
AttributeError: 'NotCompleted' object has no attribute 'distance_matrix'

**Comment by GavinHuttley** [comment]
hi!

`aligned` is a `NotCompleted` object that has been returned by the `aa_aligner()` because there was a failure in that step.

If you print `aligned` it will contain the traceback of the error that occurred. Please post that here. Can you also please post the definition of `aa_aligner()`.

**Comment by GavinHuttley** [comment]
@theo-allnutt-bioinformatics
I'm going to make a guess here. Based on the name of your script, it looks like your different alignments may be of different sets of paralogs. This indicates the sequence names may differ between the alignments. If your `aa_aligner()` is an instance of the `progressive_align()` app, then you need to do the following
```python
aa_aligner = progressive_align("protein", unique_guides=True)
```

**Comment by GavinHuttley** [comment]
Fixed in 5769cd4fa18394bf76f138388bc260def276b90c
